hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/request/EdgeRequestBuilder.java

```diff
@@ -127,12 +111,40 @@ private Single<EdgeSetGroupRequest> buildEdgeRequest(
                             timeRange,
                             space,
                             neighborIds,
-                            edgesByType.get(entityType)),
-                filters));
+                            edgesSelectionsByType.get(entityType))));
+  }
+
+  private Single<Map<String, EdgeSetRequest>> getEntityTypeToEdgeSetRequest(
+      GraphQlRequestContext context, EdgeType edgeType, Set<SelectedField> edgeFields) {
+    return Observable.fromIterable(edgeFields)
+        .collect(Collectors.groupingBy(this::getEntityType, Collectors.toUnmodifiableSet()))
+        .flatMap(entry -> this.getEdgeSetRequest(context, edgeType, entry));
   }
 
-  private Map<String, Set<SelectedField>> getEdgesByType(Stream<SelectedField> edgeSetStream) {
+  private Single<Map<String, EdgeSetRequest>> getEdgeSetRequest(
+      GraphQlRequestContext context,
+      EdgeType edgeType,
+      Map<String, Set<SelectedField>> edgeFields) {
```

Comment:
in general, be careful with the naming especial with fields since they can represent any part of the query. These, if I'm following correctly represent a map of entity types to a set of edge set fields (e.g. `outgoingEdges( neighborScope: "SERVICE")` would be one entry under the service key) rather than the fields that define an actual edge.

I missed this earlier but since we group at the level of edge scope + edge type (e.g. outgoing services), that means if I had two edge sets of the same type with different filters, the filters would be ANDed. Is that desirable? If yes, can leave. If no, you'd want to avoid the grouping by type and create a new EdgeSetRequest for every edge set field and hold them as a collection rather than mapped by entity type.

Reply:
I don't see any use case for having different filters for the same set of edges. We had the same issue when we built the use case of a third-party application flow and decided to make the filters as AND across all edges defined. I think we can leave like this.

Reply:
renamed other arguments to be more meaningful

Reply:
> I don't see any use case for having different filters for the same set of edges.

Disagree - there's plenty of compelling use cases here. I want to get a set of by slow downstream services and a set of my high error rate downstream services, for example. It's not just about AND/OR - it's what the caller wants to separate e.g. downstream external APIs vs downstream internal APIs.

Reply:
I think we might eventually want to change this, but can defer until we have a need.